network/transports.go

```diff
@@ -54,6 +54,9 @@ var backOffTemplate = wait.Backoff{
 	Steps:    15,
 }
 
+// ErrTimeoutDialing when the timeout is reached after set amount of time.
+var ErrTimeoutDialing = errors.New("timed out dialing")
```

Comment:
Rather than introduce a new error maybe we can return `context.DeadlineExceeded`

Reply:
This error type is introduced to not change the final error message. See the change in line 97:

```golang
return nil, fmt.Errorf("%w after %.2fs", ErrTimeoutDialing, elapsed.Seconds())
```

By introducing this error, consumers of this package could properly do the error checking using types:


```golang
if errors.Is(err, network.ErrTimeoutDialing) {
  // do some
}
```

instead of current dodgy string checking:


```golang
if string.Contains(err.Error(), "timed out dialing")) {
  // do some
}
```